fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -1202,6 +1198,26 @@ public void finishSink(String dbName, String tableName, List<TSinkCommitInfo> co
         }
     }
 
+    /**
+     * should clean datafiles when commit snapshot error
+     * @param batchWrite
+     * @param dataFiles
+     * @param dbName
+     * @param tableName
+     */
+    private void commitSnapshot(BatchWrite batchWrite,List<TIcebergDataFile> dataFiles,String dbName, String tableName){
+        try {
+            batchWrite.commit();
+        } catch (Exception e) {
+            List<String> toDeleteFiles = dataFiles.stream()
```

Comment:
I wonder if this is the case: It actually commits success, but encounters some issue, such as network issue, and then throws an exception? Should we check if commit actually failed before we are trying to delete these files?